src/main/java/hudson/plugins/resultscache/ResultsCacheBuildWrapper.java

```diff
@@ -107,10 +107,11 @@ public void onCompleted(AbstractBuild build, @Nonnull TaskListener listener) {
         private void saveResultToCache(AbstractBuild build, TaskListener listener, String jobHash) {
             CacheServerComm cacheServer = new CacheServerComm(ResultsCacheHelper.getCacheServiceUrl(), ResultsCacheHelper.getTimeout());
             Result r = (null != build) ? build.getResult() : Result.NOT_BUILT;
-            LoggerUtil.info(listener, "(Post Build) Sending build result for this job (result: %s :: hash: %s) %n", r, jobHash);
+            Number num = (null != build) ? build.getNumber() : -1;
```

Comment:
You see, here you are getting the value of `num` from `build.getNumber()` (which returns an `int`: https://javadoc.jenkins-ci.org/hudson/model/Run.html#getNumber--) or a default integer (-1). So, that means that you won't ever have a `null` value.

Reply:
And this means you can use `int` as type for field `build_number` in class `CacheResult`.

Reply:
Oh, I got it. You need a boxed type for managing the current values in the cache. Then, I would use `Integer`.

Reply:
I originally did have them all as Integer, but the IntelliJ IDE came up with the suggestion to convert this one to Number as "Type may be primitive", and as I didn't see the harm, I let it, and changed the others for consistency.
Doesn't actually make any difference to me though, either way, I don't think.